### ISSUE TITLE
Keep image size settings in photo modules

### ISSUE THREAD

**Issue by Florrpan** [beaverbuilder/feature-requests#117]
Each time you change an image in the image module, the image size changes/reverts to "full" image size setting. It's a very weird behaviour since you work at the same size majority of the time. I.E; a 3/3 column don't need a 2k pixel image, yet, the module keeps reverting to it when changing the image.

As mentioned, this applies to all modules that handles images, and it's enough just to press edit to add an caption for the image size to change.

**The solution**
By keeping the image size settings from the previous image we avoid accidentally adding large images, and reduces the amount of work needed when switching an image.

EDIT: You could also make it possible to "lock" this setting, and only admins(not editors) could be able to change it.

An alternative could be to have a default image size around medium, and if we need a larger image, we increase it.

**Because...**
This is crazy when you let the client in and they want to change image, ending up loading gazillion of MB extra because that setting is defaulting to full size... It's not user friendly and it adds extra work.

**Comment by zackpyle** [comment]
Not just the image module. This applies to everywhere with an image selector including background images

**Comment by frank6tg** [comment]
I agree.

**Comment by AdamTheRedBaron** [comment]
Agree. This would be a time-saver and avoid problems created by clients too!

**Comment by Pross** [comment]
Ive fixed this bug today, it'll be in an update at some point. Interestingly the bug has been there since pre 1.0 versions so at least 5 years maybe more and this is the 1st time its ever been mentioned.

**Comment by zackpyle** [comment]
@Pross Jamie said on fb that it was reported in 2018?

**Comment by Pross** [comment]
> @Pross Jamie said on fb that it was reported in 2018?

No, the request to have a default size when you add an image was suggested back in 2018 the bug that changes it to full after making a change in the wp modal has never been reported.

**Comment by zackpyle** [comment]
@Pross To be clear, does your fix include selecting Replace Image and it will now keep the previous size you selected?

**Comment by Pross** [comment]
> @Pross To be clear, does your fix include selecting Replace Image and it will now keep the previous size you selected?

Not entirely as all images have different sizes *however* you can now set a default that it will use when adding a new image or if you are replacing images.

**Comment by Florrpan** [comment]
> 
> 
> > @Pross To be clear, does your fix include selecting Replace Image and it will now keep the previous size you selected?
> 
> Not entirely as all images have different sizes _however_ you can now set a default that it will use when adding a new image or if you are replacing images.

Majority of images have the same sizes since we have a clear image (3-4 sizes), especially if we just switch the image out for another. If the next image we change to doesn't have the same size the closest size to it should be picked or it can revert to the default size.

If I picked "large" for that module on that page, then change image, I preferably want it to stay at "large", not defaulting to another size. I hope this is the intended behaviour. 

PS: pretty sure I've reported this before.. long ago.
